### PR TITLE
116 experts list filter fix

### PR DESCRIPTION
**Pull request by DelFlumen**
develop

## GitHub Board

**Issue link**

[#116 ]( )

## Summary of change

- [+]  Created FilterForm with logic and rough markup
- [+]  Added FilterForm to ExpertsView

### CODE REVIEW

File: src/lib/components/FilterForm.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  [key: string]: boolean;
+};
+
+export interface IInitLocalState {
+  [key: string]: boolean;
+}
+
+export interface ICheckboxes {
+  [key: number]: boolean;
+}
```
Comment (calster00) [inline src/lib/components/FilterForm.tsx:22]:
```suggestion
export interface ICheckboxes {
  [key: string]: boolean;
}
```
you could use just one interface here, with string keys to match initial state

File: src/modules/experts/store/expertsSlice.ts [hunk 2]
```diff
@@ -122,14 +134,28 @@ export const expertsSlice = createSlice({
...
+    setExpertsRegionsFilter: (state, action: PayloadAction<IFilter>) => {
+      if (state.experts.filters) {
+        state.experts.filters.REGIONS = action.payload;
+      }
+    },
+    setExpertsDirectionsFilter: (state, action: PayloadAction<IFilter>) => {
+      const value = action.payload;
+      if (state.experts.filters) {
+        state.experts.filters.DIRECTIONS = {
+          ...state.experts.filters.DIRECTIONS,
```
Comment (calster00) [inline src/modules/experts/store/expertsSlice.ts:146]: can't see what you spread here. shouln't this reducer be the same as `setExpertsRegionsFilter`?

File: src/modules/experts/store/expertsSlice.ts [hunk 3]
```diff
@@ -68,7 +72,14 @@ const initialState: IExpertsState = {
       loading: LoadingStatusEnum.idle,
       error: null,
     },
-    filters: [],
+    filters: {
+      [FilterTypeEnum.DIRECTIONS]: {
+        value: [],
+      },
+      [FilterTypeEnum.REGIONS]: {
+        value: [],
```
Comment (calster00) [inline src/modules/experts/store/expertsSlice.ts:80]:
```suggestion
        value: {},
      },
      [FilterTypeEnum.REGIONS]: {
        value: {},
```

File: package.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
     "@types/lodash": "^4.14.165",
+    "@types/lodash.debounce": "^4.0.6",
```
Comment (SergeOlabin) [inline package.json:14]: it should already be present in "@types/lodash

File: src/modules/experts/store/expertsSlice.ts (hunk 3 above)
Comment (SergeOlabin) [reply src/modules/experts/store/expertsSlice.ts:80]: seems like we are using an array of string values as a `value`, isn't it?

**Review by SergeOlabin — COMMENTED** [review]
pls check the changes with @calster00 
also, check package-json

File: src/modules/experts/store/expertsSlice.ts (hunk 3 above)
Comment (calster00) [reply src/modules/experts/store/expertsSlice.ts:80]: in `FilterForm` we assign an initial checkbox state of type `{ [key: string]: boolean; }` and later call `setExpertsRegionsFilter` with that value

File: src/modules/experts/store/expertsSlice.ts (hunk 2 above)
Comment (DelFlumen) [reply src/modules/experts/store/expertsSlice.ts:146]: It's only a draft, these reducers will be the same. In this PR only regions filter is ready.

File: src/lib/components/FilterForm.tsx (hunk 1 above)
Comment (DelFlumen) [reply src/lib/components/FilterForm.tsx]: fixed

File: package.json (hunk 4 above)
Comment (DelFlumen) [reply package.json]: fixed

**Review by SergeOlabin — APPROVED** [review]

**Review by darya42185 — APPROVED** [review]

**Pull request merged**